_plugins/auto-anchor.rb

```diff
@@ -21,7 +21,8 @@
         ## Remove double-quotes from titles before attempting to slugify
         title.gsub!('"', '')
         ## Use Liquid/Jekyll slugify filter to choose our id
-        id_prefix = '- {:#{{ "' + title + '" | slugify: "latin" }}} '
+        slug = '#{{ "' + title + '" | slugify: "latin" }}'
+        id_prefix = '- {:' + slug + ' .foo} <a href="' + slug +'" class="bar">🔗</a>'
```

Comment:
I'm not sure how Jekyll plugins work yet, but if this is dynamically run and performance matters, it should be faster to use string interpolation, e.g. `"#{title} #{slug}"` (which is the usual practice), rather than concatenation with the `+` operator.

Reply:
Jekyll is a static site render (although it has a live preview mode for development), so none of the code is run dynamically.  That said, I agree that this is getting hairy.  Using string concatenation made sense to me for the first version of this plugin when I just wanted to avoid having to escape quotes, but we keep doing more stuff, so now I agree it's time to bite the bullet and format it properly.

Reply:
@jnewbery pointed me to this in order to work on something similar for the review club.

Here it is, if helpful (don't hesitate to review/improve my regex): https://github.com/bitcoin-core-review-club/bitcoin-core-review-club.github.io/pull/147/files